tests/test_metrics.py

```diff
@@ -84,6 +85,20 @@ def test_f1_macro(self):
         self.assertEqual("f1_macro", outputs[0]["score"]["global"]["score_name"])
         self.assertEqual("f1_macro", outputs[0]["score"]["instance"]["score_name"])
 
+    def test_f1_weighted(self):
+        metric = F1Weighted()
+        references = [["cat"], ["dog"], ["dog"], ["dog"], ["cat"], ["cat"]]
+        predictions = ["cat", "cat", "dog", "dog", "cat", "cat"]
+        # recall class 'dog'  = 2/3  = 0.666        precision= 2/2 = 1    f1 = 0.8
+        # recall class 'cat'  = 3/3  = 1            precision= 3/4 = 0.75 f1 = 0.857142857143
+        # weighted f1 = (0.5 * 0.8) + (0.5 * 0.85714285) = 0.82857142
+        global_target = 0.82857142
```

Comment:
I think you should change the test , to ensure f1_weighted is different than f1_maco.